shared/src/commonMain/kotlin/com/hedvig/embarkx/util/Stack.kt

```diff
@@ -0,0 +1,22 @@
+package com.hedvig.embarkx.util
+
+class Stack<T> {
+    private val inner = ArrayDeque<T>()
+
+    fun pop(): T {
+        val ret = peek()
+        inner.removeLast()
+        return ret
+    }
+
+    fun push(value: T) {
+        inner.add(value)
+    }
+
+    fun peek(): T {
+        if (inner.isEmpty()) {
+            throw EmptyStackException()
+        }
+        return inner.last()
```

Comment:
Maybe make use of our convenient `orNull` alternatives? :D
```suggestion
        return inner.lastOrNull() ?: throw EmptyStackException()
```